app/Main.hs

```diff
@@ -22,10 +23,14 @@ main = do
 
     runMigration migrateAll
 
-    -- prepare users, insert users, get entities. reminder:
-    -- Section { sectionUsername :: [String], sectionParagraphs :: [[String]] }
-    let users = map (\section -> User $ sectionUsername section) sections :: [User]
-    _ <- insertMany users
+    -- insert user and pagaraph in one go
+    forM_ sections $ \section -> do
+      let user = User $ sectionUsername section
+      userId <- insert user
+      forM_ (sectionParagraphs section) $ (\p -> do
```

Comment:
I know, I was hoping that would make it more readable for untrained eyes as far as where the lambda starts and ends, but it might just be throwing people off